.github/workflows/test/issues.test.js

```diff
@@ -0,0 +1,98 @@
+import { describe, expect, it, vi, beforeEach } from "vitest";
+import { getIssueNumber } from "../src/issues.js";
+import { createMockGithub, createMockCore } from "./mocks.js";
+
+const mockGithub = createMockGithub();
+const mockCore = createMockCore();
+
+describe("getIssueNumber", () => {
+  beforeEach(() => {
+    // Reset mocks
+    vi.clearAllMocks();
+  });
+  it("should return NaN when head_sha is missing", async () => {
+    // Call function
+    const result = await getIssueNumber({
+      head_sha: "",
+      github: mockGithub,
+      core: mockCore,
+    });
+
+    // Verify result
+    expect(result.issueNumber).toBeNaN();
+
+    // Verify appropriate message was logged
+    expect(mockCore.info).toHaveBeenCalledWith(
+      "No head_sha found in check run",
+    );
+  });
+
+  it("should handle multiple PRs for the same commit", async () => {
+    // Mock multiple PRs found
+    mockGithub.rest.search.issuesAndPullRequests.mockResolvedValue({
+      data: {
+        total_count: 2,
+        items: [
+          { number: 123, html_url: "https://github.com/pr/123" },
+          { number: 456, html_url: "https://github.com/pr/456" },
+        ],
+      },
+    });
+
+    // Call function
+    const result = await getIssueNumber({
+      head_sha: "abc",
+      github: mockGithub,
+      core: mockCore,
+    });
+
+    // Verify result uses first PR
```

Comment:
Why correct to use first PR?

Reply:
Multiple PRs happens when users create multiple PRs based on the same commit sha. It's an uncommon user case and usually users should keep only one of them open. For now, just return the first PR and log warnings.  We could revisit this logic when there are more scenarios using this API in the future.